File: justjavastaff/src/main/java/com/marknkamau/justjavastaff/ui/main/MainActivity.java
package com.marknkamau.justjavastaff.ui.main;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.v7.widget.DividerItemDecoration;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ProgressBar;
import android.widget.Toast;

import com.marknkamau.justjavastaff.JustJavaStaffApp;
import com.marknkamau.justjavastaff.R;
import com.marknkamau.justjavastaff.data.network.OrdersRepository;
import com.marknkamau.justjavastaff.models.Order;
import com.marknkamau.justjavastaff.ui.MenuBarActivity;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import timber.log.Timber;

public class MainActivity extends MenuBarActivity implements MainView {
    @BindView(R.id.rv_orders)
    RecyclerView rvOrders;
    @BindView(R.id.pb_loading)
    ProgressBar pbLoading;

    private MainActivityPresenter presenter;
    private OrdersAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);

        SharedPreferences sharedPreferences = ((JustJavaStaffApp) getApplication()).getPreferences();
        OrdersRepository ordersRepository = ((JustJavaStaffApp) getApplication()).getOrdersRepository();
        presenter = new MainActivityPresenter(this, sharedPreferences, ordersRepository);

        rvOrders.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false));
        rvOrders.addItemDecoration(new DividerItemDecoration(this, LinearLayoutManager.VERTICAL));

        adapter = new OrdersAdapter(this);
        adapter.setOnClickListener(new OrdersAdapter.OnClickListener() {
            @Override
            public void onClick(Order element) {
                Toast.makeText(MainActivity.this, element.getOrderId(), Toast.LENGTH_SHORT).show();
            }
        });

        rvOrders.setAdapter(adapter);

        presenter.getOrders();
    }

    @Override
    public void displayMessage(String message) {
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
    }

    @Override
    public void displayAvailableOrders(List<Order> orders) {
        pbLoading.setVisibility(View.GONE);
        adapter.setItems(orders);
    }

    @Override
    public void displayNoOrders() {
        Toast.makeText(this, "There are no orders", Toast.LENGTH_SHORT).show();
        pbLoading.setVisibility(View.GONE);
    }
}

File: justjavastaff/src/main/java/com/marknkamau/justjavastaff/ui/main/MainActivity.kt
package com.marknkamau.justjavastaff.ui.main

import android.content.Intent
import com.google.android.material.tabs.TabLayout

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import android.os.Bundle
import com.marknkamau.justjavastaff.JustJavaStaffApp

import com.marknkamau.justjavastaff.R
import com.marknkamau.justjavastaff.ui.BaseActivity
import com.marknkamau.justjavastaff.ui.login.LogInActivity
import com.marknkamau.justjavastaff.ui.orders.OrdersFragment
import com.marknkamau.justjavastaff.ui.payments.PaymentsFragment
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setSupportActionBar(toolbar)

        val auth = (application as JustJavaStaffApp).auth
        if (!auth.isSignedIn()) {
            startActivity(Intent(this@MainActivity, LogInActivity::class.java))
            finish()
        }

        val sectionsPagerAdapter = SectionsPagerAdapter(supportFragmentManager)
        container.adapter = sectionsPagerAdapter

        container.addOnPageChangeListener(TabLayout.TabLayoutOnPageChangeListener(tabs))
        tabs.addOnTabSelectedListener(TabLayout.ViewPagerOnTabSelectedListener(container))
    }

    inner class SectionsPagerAdapter(fm: androidx.fragment.app.FragmentManager) : androidx.fragment.app.FragmentPagerAdapter(fm) {
        override fun getItem(position: Int): androidx.fragment.app.Fragment {
            return when (position) {
                0 -> OrdersFragment()
                1 -> PaymentsFragment()
                else -> OrdersFragment()
            }
        }

        override fun getCount() = 2
    }

}
